src/main/java/org/folio/circulation/resources/DeclareLostResource.java

```diff
@@ -77,4 +89,32 @@ private void declareLost(RoutingContext routingContext) {
       loan -> singleValidationError("The item is already declared lost",
         "itemId", loan.getItemId()));
   }
+
+  private CompletableFuture<Result<Loan>> createNote(Clients clients, Loan loan) {
+    final NotesRepository notesRepo = new NotesRepository(clients);
+    final NoteTypesRepository noteTypesRepo = new NoteTypesRepository(clients);
+
+    return noteTypesRepo.findByName("General note")
+      .thenApply(this::refuseIfNoteTypeNotFound)
+      .thenApply(r -> r.map(CollectionUtil::firstOrNull))
+      .thenCompose(r -> r.after(noteType -> notesRepo.create(createNote(noteType, loan))))
+      .thenApply(r -> r.map(notUsed -> loan));
+  }
+
+  private NoteRepresentation createNote(NoteType noteType, Loan loan) {
+    return new NoteRepresentation(NoteRepresentation.builder()
+      .withTitle(NOTE_MESSAGE)
+      .withTypeId(noteType.getId())
+      .withContent(NOTE_MESSAGE)
+      .withDomain(NOTE_DOMAIN)
+      .withLinks(NoteLink.from(loan.getUserId(), NoteLinkType.USER.getValue())));
+  }
+
+  private Result<MultipleRecords<NoteType>> refuseIfNoteTypeNotFound(
+    Result<MultipleRecords<NoteType>> noteTypeResult) {
+
+    return noteTypeResult.failWhen(
+      notes -> Result.succeeded(notes.getRecords().isEmpty()),
+      notes -> singleValidationError("No General note type found", "noteTypes", null));
+  }
```

Comment:
This code seems to be the same as in DeclareClaimedReturnedItemAsMissingResource.java

Could they be consolidated in some way?

Reply:
I was able to consolidate this code into the AbstractClaimedReturnedResource class.